mule-extensions-soap-api/src/main/java/org/mule/runtime/extension/api/soap/message/DispatcherResponse.java

```diff
@@ -18,10 +20,12 @@
 
   private final String contentType;
   private final InputStream content;
+  private final Map<String, ? extends List<String>> headers;
 
-  public DispatcherResponse(String contentType, InputStream content) {
+  public DispatcherResponse(String contentType, InputStream content, Map<String, ? extends List<String>> headers) {
```

Comment:
to be able to:

* Add more information about the response without the need to overload the method nor break backwards compatibility
* Being able to add handy methods to use the response and handle mule internals, such as streaming or anything else we may need to

As a side note, check every single Http framework out there, request and response are always treated as their own objects